vickyctl/src/error.rs

```diff
@@ -0,0 +1,54 @@
+use std::fmt::{Debug, Display, Formatter};
+use yansi::Paint;
+
+#[derive(Debug)]
+pub enum Error {
```

Comment:
You introduced your own error type here and derived manually from it.
In vicky, we use https://docs.rs/thiserror/latest/thiserror/.

Is there a reason for an own error implementation here?

Reply:
Yes there is. thiserror doesn't give me enough granularity over the error messages the way I would like there to be.